frontend/pages/admin/IntegrationsPage/components/AddIntegrationModal/AddIntegrationModal.tsx

```diff
@@ -49,13 +48,8 @@ const CreateIntegrationModal = ({
 
   return (
     <Modal title={"Add integration"} onExit={onCancel} className={baseClass}>
-      {testingConnection ? (
```

Comment:
This is the bug fix. It was re-rendering the UI `IntegrationForm` after `testingConnection` which cleared the states. Moved this Spinner onto `IntegrationForm` so the states won't be cleared